File: microservices/investment/src/main/java/com/illunex/invest/investment/persistence/entity/Evaluate.java
package com.illunex.invest.investment.persistence.entity;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "evaluate")
public class Evaluate {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long idx;
    Long companyIdx;
    Long vcCompanyIdx;
    String product;
    String company;
    String imgUrl;
    LocalDateTime createDate;
    LocalDateTime requestDate;
    LocalDateTime completeDate;
    LocalDateTime startupConfirmDate;
    String scale;
    String status;
    String content;
    Integer averageScore;
    Boolean deleted;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name="template_idx")
    EvaluateReviewItemTemplate template;

    @OneToMany(mappedBy = "evaluate", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    List<EvaluateJudge> judgeList;

}

File: microservices/invest/src/main/kotlin/com/illunex/invest/invest/persistence/entity/Evaluate.kt
package com.illunex.invest.invest.persistence.entity

import javax.persistence.*


@Entity
@Table(name = "evaluate")
data class Evaluate(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var idx: Long? = null,
    var companyIdx: Long? = null,
    var vcCompanyIdx: Long? = null,
    var product: String? = null,
    var company: String? = null,
    var imgUrl: String? = null,
    var updateDate: String? = null,
    var scale: String? = null,
    var status: String? = null,
    var content: String? = null,
    var score: Int? = 0,

    @OneToMany(mappedBy = "evaluate", cascade = [CascadeType.ALL])
    var judgeList: List<EvaluateJudge>?,

    @OneToOne(cascade = [CascadeType.ALL], fetch = FetchType.LAZY) @JoinColumn(name = "category_idx")
    var category: EvaluateReviewItemCategory?
) {
    constructor() : this(null, null, null, null, null, null, null, null, null, null, null, null, null )
}
